toolbox/relnotes/main.go

```diff
@@ -0,0 +1,606 @@
+	releasePRs := make([]int, 0)
+	for _, pr := range commitPRs {
+		if u.HasLabel(issueMap[pr], "release-note") {
+			releasePRs = append(releasePRs, pr)
+		}
+	}
+
+	// Generating release note...
+	log.Printf("Generating release notes...")
+
+	prFile, err := os.Create(prFileName)
+	if err != nil {
+		log.Printf("failed to create release note file %s: %s", prFileName, err)
+		return
+	}
+
+	// Bootstrap notes for minor (new branch) releases
+	if *full || u.IsVer(releaseTag, "dotzero") {
+		draftURL := u.GithubRawURL + *owner + "/features/master/" + *branch + "/release-notes-draft.md"
+		changelogURL := u.GithubRawURL + *owner + "/" + *repo + "/master/CHANGELOG.md"
+		minorRelease(prFile, releaseTag, draftURL, changelogURL)
+	} else {
+		patchRelease(prFile, startTag, releasePRs, issueMap)
+	}
+
+	prFile.Close()
+
+	// Start generating markdown file
+	log.Printf("Preparing layout...")
+
+	mdFile, err := os.Create(*mdFileName)
+	if err != nil {
+		log.Printf("failed to create release note markdown file %s: %s", *mdFileName, err)
+		return
+	}
+
+	// Create markdown file body with hardcoded kubernetes URLs
+	docURL := "https://docs.k8s.io"
+	exampleURL := "https://releases.k8s.io/" + *branch + "/examples"
+	createBody(client, mdFile, releaseTag, *branch, docURL, exampleURL, *releaseTars)
+
+	// Copy (append) the pull request notes into the output markdown file
+	prFile, _ = os.Open(prFileName)
+	_, err = io.Copy(mdFile, prFile)
+	if err != nil {
+		log.Printf("failed to copy from PR file to release note markdown file: %s", err)
+	}
+	err = mdFile.Sync()
+	if err != nil {
+		log.Printf("failed to copy from PR file to release note markdown file: %s", err)
+	}
+
+	prFile.Close()
+
+	if *preview {
+		// If in preview mode, get the pending PRs
+		err = getPendingPRs(client, mdFile, *owner, *repo, *branch)
+		if err != nil {
+			log.Printf("failed to get pending PRs: %s", err)
+			return
+		}
+	}
+	mdFile.Close()
+
+	if *htmlizeMD {
+		// Make users and PRs linkable
+		// Also, expand anchors (needed for email announce())
+		k8sGithubURL := "https://github.com/kubernetes/kubernetes"
+		_, err = u.Shell("sed", "-i", "-e", "s,#\\([0-9]\\{5\\,\\}\\),[#\\1]("+k8sGithubURL+"/pull/\\1),g",
+			"-e", "s,\\(#v[0-9]\\{3\\}-\\),"+k8sGithubURL+"/blob/master/CHANGELOG.md\\1,g",
+			"-e", "s,@\\([a-zA-Z0-9-]*\\),[@\\1](https://github.com/\\1),g", *mdFileName)
+
+		if err != nil {
+			log.Printf("failed to htmlize markdown file: %s", err)
+			return
+		}
+	}
+
+	if *preview {
+		// If in preview mode, get the current CI job status
+		// We do this after htmlizing because we don't want to update the
+		// issues in the block of this section
+		err = getCIJobStatus(*mdFileName, *branch, *htmlizeMD)
+		if err != nil {
+			log.Printf("failed to get CI status: %s", err)
+			return
+		}
+	}
+
+	if *htmlFileName != "" {
+		err = createHTMLNote(*htmlFileName, *mdFileName)
+		if err != nil {
+			log.Printf("failed to generate HTML release note: %s", err)
+		}
+	}
+
```

Comment:
At the end, you may want to print something like "Successfully generated release-note"

Reply:
Done.